centrifuge/src/main/java/io/github/centrifugal/centrifuge/Client.java

```diff
@@ -284,11 +281,25 @@ private void scheduleReconnect() {
         });
     }
 
-    private void sendSubscribeSynchronized(String channel, String token) {
-        Protocol.SubscribeRequest req = Protocol.SubscribeRequest.newBuilder()
-                .setChannel(channel)
-                .setToken(token)
-                .build();
+    private void sendSubscribeSynchronized(String channel, boolean recover, StreamPosition streamPosition, String token) {
+
+        Protocol.SubscribeRequest req = null;
+
+        if (recover) {
+            req = Protocol.SubscribeRequest.newBuilder()
+                    .setEpoch(streamPosition.getEpoch())
+                    .setGen(streamPosition.getGen())
```

Comment:
I suggest to not use `seq` and `gen` fields at all in this library. Only work with `offset` field and reflect in readme that recovery works with Centrifugo >= 2.5.0 with `v3_use_offset` option set to `true`.